### PR TITLE
Rename delete into abandon

### PR DESCRIPTION
**Pull request by Paula-Kli**
since in squeak abandon is used we renamed delete in abandon
it is the the point where one can make clean up before finally deleting an object

### CODE REVIEW

**Review by linusha — APPROVED** [review]

**Review by T4rikA — APPROVED** [review]

File: lively.morphic/morph.js [hunk 1]
```diff
@@ -1641,7 +1641,7 @@ export class Morph {
     return this;
   }
```
Comment (frcroth) [inline lively.morphic/morph.js:1643]:
```suggestion
// We do not have access to the JS VM, this method does not interact with the garbage collector and may not result in the actual deletion of the object. However, when an object is abandoned, no properties are guaranteed.
```

File: lively.morphic/morph.js [hunk 2]
```diff
@@ -1641,7 +1641,7 @@ export class Morph {
...
 
-  delete () {
+  abandon () {
     this.emptyComments();
```
Comment (frcroth) [inline lively.morphic/morph.js:1646]: Maybe remove connections

Comment (Paula-Kli) [reply lively.morphic/morph.js:1646]: we don't want to do this

File: lively.morphic/morph.js [hunk 3]
```diff
@@ -1641,7 +1641,8 @@ export class Morph {
     return this;
   }
 
-  delete () {
+  abandon () {
+    // We do not have access to the JS VM, this method does not interact with the garbage collector and may not result in the actual deletion of the object. However, when an object is abandoned, no properties are guaranteed.
```
Comment (SilvanVerhoeven) [inline lively.morphic/morph.js:1645]: Maybe change to: `// Use this method to signal the wish to permanently delete this object. Overwrite this method to clean up resources on its deletion. We do not have access to the JS VM, this method does not interact with the garbage collector and does not result in the actual deletion of the object, if there are still left over references to this object.`

**Pull request merged**